packages/server/lib/cloud/api.ts

```diff
@@ -88,18 +87,32 @@ const rp = request.defaults((params: CypressRequestOptions, callback) => {
     if (params.encrypt === true || params.encrypt === 'always') {
       const { secretKey, jwe } = await enc.encryptRequest(params)
 
+      // TODO: double check the logic below here with @tgriesser
       params.transform = async function (body, response) {
-        if (response.headers['x-cypress-encrypted'] || params.encrypt === 'always' && response.statusCode < 500) {
+        const { statusCode } = response
+
+        // response is valid and we are encrypting
+        if (response.headers['x-cypress-encrypted'] || params.encrypt === 'always') {
           let decryptedBody
 
+          // TODO: if body is null/undefined throw a custom error
```

Comment:
We should log issues for these TODO or they likely will get forgotten

Reply:
i will address/remove them all by the time this PR is done... they are notes in the places that I still have work to finish... the PR isn't done, I moved it to draft

Reply:
FWIW, this shouldn't ever be a valid scenario. Either we send a body or we don't send the response header indicating

Reply:
preflight wouldnt but thats not the only situation it could occur in